python-frontend/src/main/java/org/sonar/python/semantic/FunctionSymbolImpl.java

```diff
@@ -82,33 +87,39 @@
   }
 
   public FunctionSymbolImpl(SymbolsProtos.FunctionSymbol functionSymbolProto) {
-    this(functionSymbolProto, false);
+    this(functionSymbolProto, false, functionSymbolProto.getValidForList());
   }
 
   public FunctionSymbolImpl(SymbolsProtos.FunctionSymbol functionSymbolProto, boolean insideClass) {
-    super(functionSymbolProto.getName(), functionSymbolProto.getFullyQualifiedName());
+    this(functionSymbolProto, insideClass, functionSymbolProto.getValidForList());
+  }
+
+  public FunctionSymbolImpl(SymbolsProtos.FunctionSymbol functionSymbolProto, boolean insideClass, List<String> validFor) {
```

Comment:
It's a bit weird that we need this `validFor` parameter, to me the `SymbolsProtos.FunctionSymbol` should probably contain it, even in case of overloaded method/function.
It might make the protobuf serialization a bit heavier for no gain though, so I'm not entirely sure it's worth doing. Wdyt?

Reply:
I would prefer it to be done in the serialization too.
I think for the time being we can keep it in the deserialization, and maybe change this in another PR